spec/system/navigation_rendering_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+describe "Navigation Rendering", type: :system do
+  let(:theme) do
+    parent_theme = Fabricate(:theme, name: "Parent Theme")
+    component = Fabricate(:theme, name: "Category Sidebar Navigation", component: true)
+    parent_theme.set_default!
+  end
+
+  before do
+    theme
+    upload_theme_component
+    @category = Fabricate(:category, name: "Test Category")
+    @topic = Fabricate(:topic, category: @category)
+    @post = Fabricate(:post, topic: @topic)
+  end
```

Comment:
using `let/fab`, or `let!` in this case, is preferred in specs over instance variables:

```rb
  let!(:category) { Fabricate(:category, name: "Test Category") }
  let!(:topic) { Fabricate(:topic, category:) }
  let!(:post) { Fabricate(:post, topic:) }
  let!(:theme) do
    parent_theme = Fabricate(:theme, name: "Parent Theme")
    # is the component below used?
    component = Fabricate(:theme, name: "Category Sidebar Navigation", component: true)
    parent_theme.set_default!
  end

  before { upload_theme_component }
```